src/modules/FioAddress/action.js

```diff
@@ -11,6 +11,31 @@ import { getExchangeDenomination } from '../Settings/selectors'
 import { getFioWallets } from '../UI/selectors'
 import type { BuyAddressResponse } from './reducer'
 
+export const setFioWalletByFioAddress = (fioAddressToUse: string) => async (dispatch: Dispatch, getState: GetState) => {
+  const wallets: EdgeCurrencyWallet[] = getFioWallets(getState())
+  const fioAddress = fioAddressToUse.toLowerCase()
+  if (wallets != null) {
+    for (const wallet: EdgeCurrencyWallet of wallets) {
+      const fioAddresses: string[] = await wallet.otherMethods.getFioAddressNames()
+      if (fioAddresses.length > 0) {
+        for (const address of fioAddresses) {
+          if (address.toLowerCase() === fioAddress) {
+            return dispatch({
+              type: 'FIO/FIO_WALLET_BY_ADDRESS',
+              data: { wallet }
+            })
+          }
+        }
+      }
+    }
+  } else {
+    return dispatch({
+      type: 'FIO/FIO_WALLET_BY_ADDRESS',
```

Comment:
This redux state should simply be local component State inside of `FioPendingRequestDetailsComponent`